src/main/java/org/jenkinsci/test/acceptance/docker/fixtures/GitLabContainer.java

```diff
@@ -0,0 +1,217 @@
+    }
+
+
+    public int webPort() {
+        return port(443);
+    }
+
+    public String webHost() {
+        return ipBound(443);
+    }
+
+    public URL getURL() throws IOException {
+        return new URL("http://" + getIpAddress() + port());
+    }
+
+    public URL getHttpUrl() throws IOException {
+        String url = "http://" + ipBound(80) + ':' + port(80);
+        return new URL(url);
+    }
+
+    /** URL visible from the host. */
+    public String getRepoUrl() {
+        return "ssh://git@" + host() + ":" + port() + REPO_DIR;
+    }
+
+    @Deprecated
+    public String getRepoUrlInsideDocker() throws IOException {
+        return "ssh://git@" + getIpAddress() + REPO_DIR;
+    }
+
+    /**
+     * URL visible from other Docker containers.
+     * @param alias an alias for this container’s {@link #getCid} passed to {@code --link}
+     */
+    public String getRepoUrlInsideDocker(String alias) throws IOException {
+        return "ssh://git@" + alias + REPO_DIR;
+    }
+
+    public void waitForReady(CapybaraPortingLayer p) {
+        long timeout =  time.seconds(200); // GitLab starts in about 2 minutes add some headway
+        p.waitFor().withMessage("Waiting for GitLab to come up")
+                .withTimeout(Duration.ofMillis(timeout))
+                .pollingEvery(Duration.ofSeconds(2))
+                .until( () ->  {
+                    try {
+                        HttpRequest request = HttpRequest.newBuilder()
+                                .uri(getHttpUrl().toURI())
+                                .GET()
+                                .timeout(Duration.ofSeconds(1))
+                                .build();
+                        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+                        return response.body().contains("GitLab Community Edition");
+                    } catch (IOException ignored) {
+                        // we can not use .ignoring as this is a checked exception (even though a callable can throw this!)
+                        return Boolean.FALSE;
+                    }
+
+                });
+    }
+
+    public HttpResponse<String> createRepo(String repoName, String token) throws IOException {
+        try{
+            HttpRequest request = HttpRequest.newBuilder()
+                                             .uri(new URI(getHttpUrl() + "/api/v4/projects"))
+                                             .header("Content-Type", "application/json")
+                                             .header("PRIVATE-TOKEN", token)
+                                             .POST(HttpRequest.BodyPublishers.ofString("{ \"name\": \"" + repoName + "\" }"))
+                                             .build();
+            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+            return response;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
the method declares IOException yet you catch everything and throw a `RuntimeException`?

just propagate the original exception - let it buble up and out of the test cases?